Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift

```diff
@@ -35,22 +36,33 @@ public struct BPKSearchInputSummary: View {
     @Binding private var text: String
     @FocusState private var focused: Bool
     private let placeholder: String
-    private let inputPrefix: InputPrefix
+    private let inputPrefix: InputPrefix?
     private var state: State = .default
     
-    /// Creates a `BPKSearchInputSummary`.
-    ///
-    /// - Parameters:
-    ///   - placeholder: The placeholder text to display when the text field is empty.
-    ///   - inputPrefix: The prefix which would be displayed on the left of text input
-    ///   - text: The text to display in the text field.
+    private let readOnly: Bool
+    private let clearAction: ClearAction?
+    
+    public struct ClearAction {
+        public let accessibilityLabel: String
+        public let action: () -> Void
+        
+        public init(accessibilityLabel: String, action: @escaping () -> Void) {
+            self.accessibilityLabel = accessibilityLabel
+            self.action = action
+        }
+    }
+    
     public init(
         placeholder: String = "",
-        inputPrefix: InputPrefix = .icon(.search),
+        inputPrefix: InputPrefix? = nil,
+        clearAction: ClearAction? = nil,
+        readOnly: Bool = false,
```

Comment:
whats the expected behaviour if i set this to `readOnly = true` but also provide a clear action and I handle the click?
I'm thinking if this sould be handled by an enum, where you could either say `readonly` or `default or something` and that case receiving a clear action (again, non optional).
in the readOnly option, this component could show the icon but just not be clickable. this would make the API impossible to represent 'invalid' state.
if it even makes sense to show the (X) button when the field is readOnly. I couldn't find any design for the read only option on Figma. we should define this behaviour in figma first I thnk

Reply:
Basically, that's a figma for that: https://www.figma.com/design/RkSuI77vgPXkTf2SvuEvKf/Apps---Search-controls-GC?node-id=1867-58841&t=xeqerj6azCGFcs1a-4

Read only should affect only user text input ability. But we still can set the text value outside. And user still may remove the text by clicking the clear button. So, they're independent